platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/signed/SignedStateFileWriter.java

```diff
@@ -135,27 +143,142 @@ public static void writeStateFile(final Path directory, final SignedState signed
     /**
      * Write all files that belong in the signed state directory into a directory.
      *
-     * @param selfId        the id of the platform
-     * @param directory     the directory where all files should be placed
-     * @param signedState   the signed state being written to disk
-     * @param configuration the configuration used
+     * @param platformContext the platform context
+     * @param selfId          the id of the platform
+     * @param directory       the directory where all files should be placed
+     * @param signedState     the signed state being written to disk
      */
     public static void writeSignedStateFilesToDirectory(
+            @Nullable final PlatformContext platformContext,
             @Nullable NodeId selfId,
             @NonNull final Path directory,
-            @NonNull final SignedState signedState,
-            @NonNull final Configuration configuration)
+            @NonNull final SignedState signedState)
             throws IOException {
-        Objects.requireNonNull(directory, "directory must not be null");
-        Objects.requireNonNull(signedState, "signedState must not be null");
-        Objects.requireNonNull(configuration, "configuration must not be null");
+        Objects.requireNonNull(platformContext);
+        Objects.requireNonNull(directory);
+        Objects.requireNonNull(signedState);
 
         writeStateFile(directory, signedState);
         writeHashInfoFile(directory, signedState.getState());
         writeMetadataFile(selfId, directory, signedState);
         writeEmergencyRecoveryFile(directory, signedState);
         writeStateAddressBookFile(directory, signedState.getAddressBook());
-        writeSettingsUsed(directory, configuration);
+        writeSettingsUsed(directory, platformContext.getConfiguration());
+        copyPreconsensusEventStreamFiles(
+                platformContext,
+                selfId,
+                directory,
+                signedState.getState().getPlatformState().getPlatformData().getMinimumGenerationNonAncient());
+    }
+
+    /**
+     * Copy preconsensus event files into the signed state directory. These files are necessary for the platform to use
+     * the state file as a starting point. Note that starting with this only the PCES files in the state file does not
```

Comment:
Confusing grammar "Note that starting with this only the"

Reply:
```
     * Copy preconsensus event files into the signed state directory. These files are necessary for the platform to use
     * the state file as a starting point. Note: starting a node using the PCES files in the state directory does not
     * guarantee that there is no data loss (i.e. there may be transactions that reach consensus after the state
     * snapshot), but it does allow a node to start up and participate in gossip.
```